h2o-core/src/main/java/hex/ModelBuilder.java

```diff
@@ -1443,6 +1443,9 @@ public void init(boolean expensive) {
         if (_parms._check_constant_response && _response.isConst()) {
           error("_response", "Response cannot be constant.");
         }
+        if(_nclass == 1 && _response.toCategoricalVec().cardinality() == 2){
```

Comment:
this is leaking the transformed vec - it is generally an expensive operation

I think it would be better to limit this warning only to binary response (0/1) and use isBinary

Reply:
Sorry, I forgot to set this PR as a draft. 

Thank you for your suggestions. I miss the isBinary function.. Also, I will implement the -1/1 check!

Reply:
@michalk for -1/+1 case there is no easy to get a number of zeros from the rollups stats.. You can get a number of zeros from a sparse chunk only, or am I missing something?

Reply:
@maurever 

_nzCnt,       //count(X!=0) for non-NA values

you can check this value

vec.length() - _nzCnt - _naCnt - this will give you the number of zeros

Reply:
Done here: https://github.com/h2oai/h2o-3/pull/5677/commits/ed57a95ecbe17f0f46d74608a5c5b2c1dc1dc582, thank you @michalkurka for your ideas!